check_integrity.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+err() {
+    echo "$1"
+    exit 1
+}
+
+check_bash_script() {
+    local cmd="git-$1"
+    test -x "bin/$cmd" || err "bin/$cmd is either non-existent or unexecutable"
```

Comment:
I'm reviewing the file now. I'll be posting a patch in a few minutes, and I'll address that.